lint/util.py

```diff
@@ -908,30 +908,40 @@ def find_windows_python(version):
     """Find the nearest version of python and return its path."""
 
     if version:
-        # On Windows, there may be no separately named python/python3 binaries,
-        # so it seems the only reliable way to check for a given version is to
-        # check the root drive for 'Python*' directories, and try to match the
+        from . import persist
+
+        # Try to find python.exe using py.exe
+        # Try the exact version first, then the major version
+        code = r'import sys;sys.stdout.write(sys.executable)'
+        for testversion in (version, version.partition('.')[0]):
+            path = communicate(('py.exe', '-%s' % testversion), code)
+            if path:
+                if ('python.exe' in path.lower()) and can_exec(path):
+                    persist.debug('find_windows_python: <=', path)
+                    return path
+
+        # Check the root drive for 'Python*' directories, and try to match the
         # version based on the directory names. The 'Python*' directories end
         # with the <major><minor> version number, so for matching with the version
         # passed in, strip any decimal points.
         stripped_version = version.replace('.', '')
         prefix = os.path.abspath(os.path.join(
-            os.environ.get("SYSTEMROOT", "\\")[:2],
+            os.environ.get("SYSTEMDRIVE", ""),
+            os.sep,
```

Comment:
Come to think of it, you shouldn't really need the os.sep as those are arguments of os.path.join.

Reply:
@rchl But it doesn't work when there's a drive letter at the beginning.

```python
>>>os.path.join("C:", "Python")
'C:Python'
```
This causes it to use "C:\Program Files\Sublime Text 3\Python" as the `prefix`.

You could use something like this,

```python
prefix = os.path.abspath(os.path.join(
    os.environ.get('SYSTEMDRIVE'),
    os.sep,
    'Python'
))
```

This will also fix some pylint related issues.